### PR TITLE
Normalize CNAME/ALIAS/PTR value to lower-case

### PR DESCRIPTION
**Pull request by PeterDaveHello**
Related to #337

Currently, CNAME value is not normalized, but might but stored by provider in lower case, which will cause just almost the same issue as #322(one is MX but one is CNAME): found diff and update every sync, and it could be solved by normalization CNAME value.

I also add a small test, just like #337, to make sure the behavior won't appear in the future become a regression bug. Without the normalization, tests will fail as below:

```
..............................................................................................................................................................................................................................F........................................................
======================================================================
FAIL: test_cname_lowering_value (test_octodns_record.TestRecord)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pete/octodns/tests/test_octodns_record.py", line 42, in test_cname_lowering_value
    self.assertEquals(upper_record.value, lower_record.value)
AssertionError: u'GITHUB.COM' != u'github.com'
- GITHUB.COM
+ github.com

-------------------- >> begin captured logging << --------------------
Record: DEBUG: __init__: zone.name=unit.tests., type=CnameRecord, name=CnameUppwerValue
Record: DEBUG: __init__: zone.name=unit.tests., type=CnameRecord, name=CnameLowerValue
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 279 tests in 4.730s

FAILED (failures=1)

```

### CODE REVIEW

File: octodns/record/__init__.py [hunk 1]
```diff
@@ -718,7 +718,10 @@ def process(self, value):
 
 
 class CnameValue(_TargetValue):
-    pass
+
+    @classmethod
+    def process(self, value):
+        return value.lower()
```
Comment (ross) [inline octodns/record/__init__.py]: I believe the `.lower()` will need to be up line ~717 in the parent class so that it'll fix ALIAS records as well.

Comment (PeterDaveHello) [reply octodns/record/__init__.py]: I'm not sure if you prefer to further revising other parts as simply mv it to `class _TargetValue(object)` will have some other issues:

```
======================================================================
ERROR: test_value_mixin_data (test_octodns_record.TestRecord)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/pete/octodns/tests/test_octodns_record.py", line 152, in test_value_mixin_data
    'value': None
  File "/home/pete/octodns/octodns/record/__init__.py", line 365, in __init__
    self.value = self._value_type.process(data['value'])
  File "/home/pete/octodns/octodns/record/__init__.py", line 717, in process
    return value.lower()
AttributeError: 'NoneType' object has no attribute 'lower'
-------------------- >> begin captured logging << --------------------
Record: DEBUG: __init__: zone.name=unit.tests., type=AliasRecord, name=
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
```

Comment (PeterDaveHello) [reply octodns/record/__init__.py]: Or I can do a string detection in `class _TargetValue(object)` and decide to lower or not, what do you think?

Comment (ross) [reply octodns/record/__init__.py]: Yeah. I believe that's a case of where Alias is being used to test _TargetValue. Both CNAME and ALIAS could have a None value, just ALIAS has a test.

It should be fine to do an 

```python
if value:
   return value.lower()
```

There's no need to check for a string in that specific case as that'd go beyond the types of checking that octoDNS currently does (everywhere.)

Comment (PeterDaveHello) [reply octodns/record/__init__.py]: Fair enough, I also noticed PTR value also uses `_TargetValue`, added it here together(hope I'm not wrong 😓 )

Comment (PeterDaveHello) [reply octodns/record/__init__.py]: All updated and passed tests now 🎉

**Pull request merged**